tasks/git_deploy.js

```diff
@@ -35,22 +39,54 @@ module.exports = function(grunt) {
 
     function git(args) {
       return function(cb) {
-        grunt.log.writeln('Running ' + args.join(' ').green);
+        grunt.log.writeln('Running ' + args.join(' ').green + ' in ' + deployDir );
         spawn({
           cmd: 'git',
           args: args,
-          opts: {cwd: src}
+          opts: {cwd: deployDir}
         }, cb);
       };
     }
 
+    function copyDirectory( src_dir, dest_dir ){
+
+          return function(cb) {
+              grunt.log.writeln('Copying ' + src_dir + ' to ' + dest_dir );
+              //Ensure directory has trailingslash
+              if ( src_dir.substr(-1) != '/' ) {
+                  src_dir = src_dir + '/';
+              }
+              grunt.file.expand(  { 'expand': true, 'cwd' : src_dir }, '**/*' ).forEach( function( src ){
+
+                  var dest;
+
+                  if (process.platform === 'win32') {
+                      dest = path.join( dest_dir, src).replace(/\\/g, '/');
+                  } else {
+                      dest = path.join( dest_dir, src);
+                  }
+
+                  if ( grunt.file.isDir( src_dir + src ) ) {
+                      grunt.file.mkdir( dest);
+                  } else {
+                      grunt.file.copy( src_dir + src, dest );
+                  }
+              });
+
+              cb();
+          };
+      }
+
+    grunt.file.mkdir( deployDir );
+
     grunt.file.delete(path.join(src, '.git'));
 
     var done = this.async();
 
     grunt.util.async.series([
-      git(['init']),
-      git(['checkout', '--orphan', options.branch]),
+      git(['clone', options.url, '.' ]),
+      git(['checkout', '-B', options.branch]),
+      copyDirectory( src, deployDir ),
       git(['add', '--all']),
       git(['commit', '--message="' + options.message + '"']),
       git(['push', '--prune', '--force', '--quiet', options.url, options.branch])
```

Comment:
Depends - I think `--force` keeps existing behaviour in that if you encounter a merge conflict it will override the history.